apa.csl

```diff
@@ -146,243 +400,466 @@
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <macro name="archive">
+    <group delimiter=". ">
+      <group delimiter=", ">
         <choose>
-          <if variable="genre">
-            <text variable="genre" text-case="capitalize-first"/>
+          <if type="manuscript">
+            <text variable="genre"/>
           </if>
-          <else>
-            <!-- This should be localized -->
-            <text value="Data set"/>
-          </else>
         </choose>
-      </else-if>
-    </choose>
+        <group delimiter=" ">
+          <!-- Replace "archive" with "archive_collection" as that becomes available -->
+          <text variable="archive"/>
+          <text variable="archive_location" prefix="(" suffix=")"/>
+        </group>         
+      </group>
+      <group delimiter=", ">
+        <!-- Move "archive" here when "archive_collection" becomes available -->
+        <text variable="archive-place"/>
+      </group>
+    </group>
   </macro>
   <macro name="publisher">
     <choose>
-      <if type="report" match="any">
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher"/>
-        </group>
+      <if type="manuscript pamphlet" match="any">
+        <choose>
+          <if variable="archive archive_location archive-place" match="any">
+            <group delimiter=". ">
+              <group delimiter=": ">
+                <text variable="publisher-place"/>
+                <text variable="publisher"/>
+              </group>
+              <text macro="archive"/>
+            </group>
+          </if>
+          <else>
+            <group delimiter=", ">
+              <text variable="genre"/>
+              <text variable="publisher"/>
+              <text variable="publisher-place"/>
+            </group>
+          </else>
+        </choose>
       </if>
       <else-if type="thesis" match="any">
-        <group delimiter=", ">
-          <text variable="publisher"/>
-          <text variable="publisher-place"/>
+        <group delimiter=". ">
+          <group delimiter=", ">
+            <text variable="publisher"/>
+            <text variable="publisher-place"/>
+          </group>
+          <text macro="archive"/>
         </group>
       </else-if>
       <else-if type="patent">
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <choose>
-            <if variable="publisher">
-              <text variable="publisher"/>
-            </if>
-            <else>
-              <text variable="authority"/>
-            </else>
-          </choose>
+        <group delimiter=". ">
+          <group delimiter=": ">
+            <text variable="publisher-place"/>
+            <choose>
+              <if variable="publisher">
+                <text variable="publisher"/>
+              </if>
+              <else>
+                <text variable="authority"/>
```

Comment:
The example reference in the manual is:
Smith, I. M. (1988). _US.S. Patent No. 123,445._ Washington, DC: U.S. Patent and Trademark Office.

The "Washington, DC: U.S. Patent and Trademark Office" bit is intended to refer to "the official source from which the patent information can be retrieved". I actually think that `authority` is inappropriate to include there (i.e., there is no reason that "U.S. Patent and Trademark Office" should be stored in `authority`).

Reply:
@adam3smith You okay with dropping authority here entirely then?

Reply:
Wouldn't it be needed for the U.S. in U.S. Patent just as in the citation? I'm definitely OK to not use it for Washington, DC: U.S. Patent and Trademark Office. (though looking back me doing it this way may have been with Mendeley in mind?)

Reply:
Sorry, I meant to remove it from the publisher macro because authority is being used in the patent number macro above. Mendeley doesn’t use authority at all, maybe it was another client? 

One other question—patents stand out as the only type in the APA Manual that don’t use a DOI/URL in place of the publisher name/location if available. Patents aren’t mentioned in the 2012 update focused on electronic references. 

Should (1) the URL or DOI be included if no publisher is present, and (2) a URL or DOI take precedence over publisher?

Reply:
Ah, it looks like you wrote it this way with Zotero in mind—Zotero doesn’t have either publisher or publisher place on patent.

Reply:
Okay, APA Style has indicated that it is acceptable to replace the publisher/place with URL or DOI for patents consistent with other item types.